overlord/snapstate/handlers.go

```diff
@@ -1936,6 +1941,35 @@ func notifyLinkParticipants(t *state.Task, snapsup *SnapSetup) {
 	}
 }
 
+// maybeUninhibitSnap uninhibits given snap if no "uninhibit-snap" task is schedueled to run in current change.
+//
+// This is intended to be only used by "link-snap" to address the case where we
+// have pending changes without "uninhibit-snap" to avoid having snaps that are
+// never uninhibited.
+func (m *SnapManager) maybeUninhibitSnap(t *state.Task, info *snap.Info) (err error) {
+	// XXX: would this be good place to use findTasksMatchingKindAndSnap?
+	// check if we have an uninhibit-snap task later
+	chg := t.Change()
+	for _, t := range chg.Tasks() {
+		if t.Kind() != "uninhibit-snap" || t.Status().Ready() {
```

Comment:
This was just a sanity check, it should always be "not ready" if things are going as expected. Do you think it should be removed?

Reply:
No, it's fine to keep it as a sanity check I think. I asked because it didn't seem strictly necessary so I just wanted to be sure wasn't misunderstanding something.